hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/MembershipManager.java

```diff
@@ -764,7 +770,10 @@ void sendMembershipEvents(Collection<MemberImpl> currentMembers, Collection<Memb
                 // sync calls
                 node.getPartitionService().memberAdded(newMember);
                 node.getNodeExtension().onMemberListChange();
-
+                Joiner joiner = node.getJoiner();
+                if (joiner != null && joiner.getClass() == TcpIpJoiner.class) {
```

Comment:
simply `if (joiner instanceof TcpIpJoiner)`

Reply:
Done.

Reply:
I will revert this change since we don't want to remember the member addresses for the other child class joiners, `DiscoveryJoiner`. See https://github.com/hazelcast/hazelcast/pull/21552#discussion_r934305495